include/roboteam_utils/containers/state_machine.hpp

```diff
@@ -138,7 +138,7 @@ namespace rtt::collections {
          * Skips n elements, sets current to clamp(current + n, _data.size())
          * @param n Amount of elements to skip
          */
-        void skip_n(int n) noexcept {
+        void skip_n(std::size_t n) noexcept {
```

Comment:
yeah i had this initially, jesse did not agree so we changed it to `int`.

Reply:
std::size_t is clearer, int will be more likely to cause unintended conversions and overflows so I think this is better aswell.

Reply:
As in, I tried to test it and got segfaults, so that's a good reason to do so I think.